src/crimson/os/seastore/async_cleaner.h

```diff
@@ -1290,7 +1408,32 @@ class SegmentCleaner : public SegmentProvider {
       ool_segment_seq_allocator->set_next_segment_seq(seq);
     }
   }
+
+  journal_seq_t get_journal_head() const {
+    if (journal->get_type() ==
+	journal_type_t::SEGMENT_JOURNAL) {
+      return segments.get_journal_head();
+    } else {
+      return journal_head_committed;
+    }
+  }
+
+  seastar::future<> update_tail_if_needed() {
+    if (journal && journal->get_type() ==
+	journal_type_t::CIRCULARBOUNDED_JOURNAL) {
```

Comment:
Why is this behavior different for a CBJournal?  Either we need to make the behavior identical or this needs a comment explaining why there's a difference in semantics.

Reply:
Unlike existing segmented journal, CBJournal's tail needs to be updated after trimming is done because it manages tail information at the header---the tail is only updated after the trimming.

Reply:
`update_tail_if_needed()` doesn't seem to behave differently, I think it is possible to reuse `get_dirty_extents_replay_from()` and `get_alloc_info_replay_from()` instead.